### PR TITLE
Add spring config metadata

### PR DESCRIPTION
**Pull request by worldtiki**

### CODE REVIEW

File: src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
+{
+  "properties": [
+    {
+      "name": "pitchfork.forwarders.haystack.enabled",
+      "type": "java.lang.Boolean",
+      "description": "When true, creates a forwarder to a Haystack Kafka topic."
+    },
+    {
+      "name": "pitchfork.forwarders.haystack.kafka.broker-url",
```
[the remark this answers is not in the document]
Comment (worldtiki) [reply src/main/resources/META-INF/additional-spring-configuration-metadata.json]: Yeap, good point. Thanks

File: src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      "type": "java.lang.Boolean",
+      "description": "When true, logs the entire content of the span received. When false only the id is logged."
+    },
+    {
+      "name": "pitchfork.forwarders.zipkin.enabled",
+      "type": "java.lang.Boolean",
+      "description": "When true, creates a forwarder that publishes spans into a Zipkin collector."
+    },
+    {
+      "name": "pitchfork.forwarders.zipkin.host",
```
[the remark this answers is not in the document]
Comment (worldtiki) [reply src/main/resources/META-INF/additional-spring-configuration-metadata.json]: Thanks. Updated the pr to reflect that.

File: src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      "type": "java.lang.String",
+      "description": "Name of the Haystack's Kafka topic."
+    },
+    {
+      "name": "pitchfork.forwarders.logging.enabled",
+      "type": "java.lang.Boolean",
+      "description": "When true, creates a forwarder that simply logs incoming spans."
+    },
+    {
+      "name": "pitchfork.forwarders.logging.log-full-span",
```
[the remark this answers is not in the document]
Comment (worldtiki) [reply src/main/resources/META-INF/additional-spring-configuration-metadata.json]: Just the span. If there is an exception before (for example unzipping compressed requests) it won't be logged by this forwarder.

I need to spend a bit of time around this, namely adding metrics for failures/successes so it's easier to track than having to analyse logs, as well as better logging for errors.

File: pom.xml [hunk 4]
```diff
@@ -92,6 +92,14 @@
       <artifactId>zipkin-sender-okhttp3</artifactId>
     </dependency>
 
+    <!-- Compile time dependencies -->
+    <dependency>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-configuration-processor</artifactId>
+      <version>2.0.6.RELEASE</version>
```
[the remark this answers is not in the document]
Comment (worldtiki) [reply pom.xml]: I'll leave that for later.
I thought it was a straightforward upgrade but I had some issues with configuring the pipeline to handle compressed http requests.

**Comment by adriancole** [comment]
appreciate all the consideration!

**Pull request merged**